toolchain/check/context.h

```diff
@@ -45,6 +46,15 @@ struct LookupResult {
   SemIR::InstId inst_id;
 };
 
+// Information about an access
+struct AccessInfo {
+  // the instruction being accessed
+  SemIR::InstId base_id;
+
+  bool allow_private_access;
+  bool allow_protected_access;
```

Comment:
Had you considered reusing the AccessKind enum here, rather than providing separate booleans?

```suggestion
  SemIR::AccessKind allowed_access;
```

Reply:
I considered doing this. But, I'm not sure how `allowed_access` can contain information if the access is allowed for _both_ protected and private, or only needing private. 

I thought of turning `AccessKind` into a flag enum, similar to `NodeCategory`, but I felt conflicted since `AccessKind` is originally used to indicate an exclusive access in `Entry.` 

Or perhaps do you mean, like `highest_allowed_access`, so if the access is allowed for `private`, it's allowed for `protected` as well?